modules/trusted-firmware-m/Kconfig.tfm

```diff
@@ -206,6 +206,31 @@ config TFM_BL2
 	  TFM is designed to run with MCUboot in a certain configuration.
 	  This config adds MCUboot to the build - built via TFM's build system.
 
+config TFM_BUILD_NS
+	bool "Build the TF-M Non-Secure application and libraries"
+	help
+	  Instruct the TF-M build system to build the TF-M Non-Secure
+	  application and libraries.
+
+	  This option is intended for testing purposes only, since this is the
+	  easiest way to build the TF-M non-secure test application and
+	  support libraries in the zephyr build system.
+
+config TFM_APP_NS
+	bool "Use the TF-M Non-Secure application"
+	depends on TFM_BUILD_NS
```

Comment:
This is a bit confusing when you see:

```
CONFIG_TFM_BUILD_NS=y
CONFIG_TFM_APP_NS=y
```

I agree separating build and use is useful, but the current naming isn't very obvious.

Perhaps `TFM_BUILD_NATIVE_NS` and `TFM_USE_NATIVE_NS`? I don't love the `native` label but don't have a better suggestion.

If I just see this in my `prj.conf` file it's a bit more obvious what the difference is (to me anyway):

```
TFM_BUILD_NATIVE_NS=y
TFM_USE_NATIVE_NS=y
```

Reply:
> I agree separating build and use is useful

It is required, because the psa-test sample uses the zephyr application, but requires to link against the TF-M non-secure libraries.

What about this:
```
CONFIG_TFM_BUILD_NS=y
CONFIG_TFM_USE_NS_APP=y
```

Reply:
We're talking about the TF-m-tests binary with `CONIG_TFM_USE_NS_APP`, correct? In that case I suggest something like `CONFIG_TFM_USE_NS_TEST_APP` and `CONFIG_TFM_BUILD_NS_LIBRARY` so that it's more clear. Yeah, they're longer, but I think you can type it the ~5 times needed.

Reply:
Either of these is fine to me. I just think we should try to be clearer, and the existing names are ambiguous.

Reply:
Ok. I'm going with TFM_USE_NS_APP, since this binary is referred to as the TF-M non-secure application.